### PR TITLE
Sharded tests pt 3: manually set arcs

### PR DESCRIPTION
**Pull request by maackle**

### CODE REVIEW

File: crates/holochain/src/sweettest/sweet_conductor.rs [hunk 1]
```diff
@@ -22,6 +23,9 @@ pub type SignalStream = Box<dyn tokio_stream::Stream<Item = Signal> + Send + Syn
 /// A useful Conductor abstraction for testing, allowing startup and shutdown as well
 /// as easy installation of apps across multiple Conductors and Agents.
 ///
+/// SweetConductors are meant to run on a single machine. Peer discovery is
+/// explicit (TODO), and they are not expected to use a bootstrap server.
+///
```
Comment (freesig) [inline crates/holochain/src/sweettest/sweet_conductor.rs:28]: We already have tests that do explicit peer discovery though?

**Review by freesig — CHANGES_REQUESTED** [review]
This PR still needs to change the `agent_arc` map on the kitsune Space to also hold the forced arc. In fact if you changed that map you wouldn't need to directly change the database because that is what sets the arc in the database.

File: crates/holochain/src/sweettest/sweet_conductor.rs (hunk 1 above)
Comment (maackle) [reply crates/holochain/src/sweettest/sweet_conductor.rs:28]: Oh yeah I should clarify that I want to be able to discover peers in a more fine grained way than "everyone discovers everyone else"

**Comment by neonphog** [comment]
@maackle - Is this pointing at the right target?? LOC says: `+3,095 −986`

File: crates/holochain_p2p/src/test.rs [hunk 2]
```diff
@@ -124,6 +124,14 @@ impl HolochainP2pHandler for StubNetwork {
     ) -> HolochainP2pHandlerResult<()> {
         Err("stub".into())
     }
+    #[cfg(feature = "test_utils")]
+    fn handle_test_backdoor(
```
Comment (neonphog) [inline crates/holochain_p2p/src/test.rs:128]: bikeshedding, but i don't like when items bump into each other like this... i wish rustfmt defaulted to [blank_lines_lower_bound = 1](https://rust-lang.github.io/rustfmt/?version=v1.4.36&search=#blank_lines_lower_bound).

File: crates/kitsune_p2p/kitsune_p2p/src/config.rs [hunk 3]
```diff
@@ -46,6 +48,10 @@ impl Default for KitsuneP2pConfig {
             bootstrap_service: None,
             tuning_params: KitsuneP2pTuningParams::default(),
             network_type: NetworkType::QuicBootstrap,
+            #[cfg(feature = "sharded")]
+            resize_arcs: true,
+            #[cfg(not(feature = "sharded"))]
+            resize_arcs: false,
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/config.rs:54]: have we not done the work to switch from feature flag to tuning param yet?

I thought the intention was to not have to recompile to alter this behavior.

**Review by neonphog — APPROVED** [review]
nice :+1:

File: crates/holochain_p2p/src/test.rs (hunk 2 above)
Comment (maackle) [reply crates/holochain_p2p/src/test.rs:128]: @neonphog I was curious about this too, I ran our codebase through fmt with that option, and here's what it looks like. I think it's a bit overzealous in terms of what it thinks of as an "item". https://github.com/holochain/holochain/compare/blank-lines-lower-bound-1

File: crates/kitsune_p2p/kitsune_p2p/src/config.rs (hunk 3 above)
Comment (maackle) [reply crates/kitsune_p2p/kitsune_p2p/src/config.rs:54]: This is a bit of a placeholder. This "feature" isn't even hooked up anywhere else, kind of just rolling with what we had before. Note that this is just changing the default value of the config. I could hook this up more properly or leave a comment about it, just not sure how far to go with it.

File: crates/holochain_p2p/src/test.rs (hunk 2 above)
Comment (neonphog) [reply crates/holochain_p2p/src/test.rs:128]: hah - yeah, that's no good : )

File: crates/kitsune_p2p/kitsune_p2p/src/config.rs (hunk 3 above)
Comment (neonphog) [reply crates/kitsune_p2p/kitsune_p2p/src/config.rs:54]: for sure, I'm fine with the work to be done in a different effort, just don't want to forget about it : )

File: crates/holochain/src/sweettest/sweet_conductor.rs (hunk 1 above)
Comment (maackle) [reply crates/holochain/src/sweettest/sweet_conductor.rs:28]: BTW this is done now in the next PR

File: crates/kitsune_p2p/kitsune_p2p/src/config.rs (hunk 3 above)
Comment (freesig) [reply crates/kitsune_p2p/kitsune_p2p/src/config.rs:54]: Probably want to remove this now that we do have a PR for this?

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs [hunk 4]
```diff
@@ -372,9 +367,12 @@ async fn update_single_agent_info(
         network_type,
         mdns_handles,
         bootstrap_service,
+        resize_arcs,
     } = input;
 
-    update_arc_length(evt_sender, space.clone(), &mut arc).await?;
+    if resize_arcs {
```
Comment (freesig) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs:373]: I guess this was already done?

**Review by freesig — APPROVED** [review]
Looks good. Might conflict with #943

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs (hunk 4 above)
Comment (maackle) [reply crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs]: Ah sorry @freesig didn't realize this was exactly what you meant by plumbing this config through

**Pull request merged**